src/api/tree/mutate/downloadVariable.ts

```diff
@@ -0,0 +1,74 @@
+import { ResponseParser, apiFactory } from "~/api/apiFactory";
+
+import { WorkflowVariableDownloadSchema } from "../schema/workflowVariable";
+import { forceLeadingSlash } from "../utils";
+import { useApiKey } from "~/util/store/apiKey";
+import { useMutation } from "@tanstack/react-query";
+import { useNamespace } from "~/util/store/namespace";
+import { useToast } from "~/design/Toast";
+import { useTranslation } from "react-i18next";
+
+const blobResponseParser: ResponseParser = async ({ res, schema }) => {
+  const varBlob = await res.blob();
+  const headers = Object.fromEntries(res.headers);
+  return schema.parse({ blob: varBlob, headers });
+};
+
+export const getVarBlob = apiFactory({
```

Comment:
My first intuition was going to a query as well, but it should be a mutation because it is actively requested by the user via a click interaction. And queries (as in useQuery) are not meant to be triggered like that.